src/reconcile/rec_write.c

```diff
@@ -1231,36 +1244,42 @@ __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
 			if (WT_TXNID_LT(txnid, min_txn))
 				min_txn = txnid;
 
```

Comment:
To your specific point about `max_txn == UINT64_MAX` -- I think you're suggesting that local variables have no semantic meaning, so the compiler can completely ignore them and do multiple loads from memory.

I don't think is the case (and would break other code, e.g., I remember a bug in `__wt_random` that was resolved by reading the shared state into a local variable).

Even if we do need to worry about it, making `WT_UPDATE->txnid` into a `volatile` should be sufficient, right?  We don't need to order subsequent reads with respect to the read of `upd->txnid`, just make sure it is read once.